File: programmers/python/같은_숫자는_싫어.py
# https://programmers.co.kr/learn/courses/30/lessons/12906
def solution(arr):
    answer = []
    
    temp = arr[0]    
    for i in arr[1:]:
        if temp != i:
            answer.append(temp)
            temp = i
            
    if temp == arr[-1]:
        answer.append(arr[-1])
            
    return answer

File: programmers/java/같은_숫자는_싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        int size = 0;
        int pre = -1;

        for(int i: arr){
            if(pre != i)
                answer.add(i);
            pre = i;
        }

        int[] answer_arr = new int[answer.size()];
        
        for(int i: answer)
            answer_arr[size++] = i;
        
        return answer_arr;
        
    }
}